mii/client.py

```diff
@@ -156,6 +166,48 @@ def destroy_session(self, session_id):
             client.destroy_session(session_id)
 
 
+class MIINonPersistentClient():
+    def __init__(self, task, deployment_name):
+        self.task = task
+        self.deployment_name = deployment_name
+
+    def query(self, request_dict, **query_kwargs):
+        task_methods = GRPC_METHOD_TABLE[self.task]
+        inference_pipeline = mii.non_persistent_models[self.deployment_name][0]
+
+        if self.task == Tasks.QUESTION_ANSWERING:
+            if 'question' not in request_dict or 'context' not in request_dict:
+                return
+
+            return task_methods.run_inference(inference_pipeline,
+                                              request_dict,
+                                              query_kwargs,
+                                              True)
+
+        elif self.task == Tasks.CONVERSATIONAL and 'text' in request_dict and 'conversation_id' in request_dict and 'past_user_inputs' in request_dict and 'generated_responses' in request_dict:
+            kwargs = {}
+            text = request_dict['text']
+            conversation_id = request_dict['conversation_id']
+            past_user_inputs = request_dict['past_user_inputs']
+            generated_responses = request_dict['generated_responses']
+            conv = Conversation(text=text,
+                                conversation_id=conversation_id,
+                                past_user_inputs=past_user_inputs,
+                                generated_responses=generated_responses,
+                                **kwargs)
+            args = (conv, )
```

Comment:
I wonder if we could refactor the Conversation task methods so we could re-use the code here:
https://github.com/microsoft/DeepSpeed-MII/blob/dc5ab44dfa48ae9f0a99b356e96c8849c0c78aea/mii/method_table.py#L192

Maybe bring the `conv = Conversation(...)` bit out into it's own function called `create_conversation` and use `getattr(request, "conversation_id"), getattr(request, "past_user_inputs"), ...` so that the code works with the proto message and the query dictionary.